### PR TITLE
Feature/ontology import

### PR DESCRIPTION
**Pull request by michael-wirth**

### CODE REVIEW

File: app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/X28ObligationImportTask.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.cloud.task.configuration.EnableTask;
+
+import ch.admin.seco.service.reference.integration.x28.ontology.importer.config.X28Properties;
+
+@SpringBootApplication
+@EnableTask
+@EnableBatchProcessing
+@EnableBinding(Source.class)
+@EnableConfigurationProperties(X28Properties.class)
+public class X28ObligationImportTask {
```
Comment (birom) [inline app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/X28ObligationImportTask.java]: There is typo here. This should be X28OntologyImporTask

File: app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/config/X28OntologyImportTaskConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        this.x28OntologyDataFileMessageSource = x28OntologyDataFileMessageSource;
+        this.messageBrokerOutputChannel = output;
+    }
+
+    @Bean
+    Job x28ImportJob(
+            StaxEventItemReader<Row> xmlFileReader,
+            ItemProcessor<Row, CreateOccupationLabelDto> occupationLabelDtoItemProcessor,
+            X28OntologyWriter x28JobAdWriter) {
+        return jobBuilderFactory.get("x28-jobad-xml-import")
```
Comment (birom) [inline app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/config/X28OntologyImportTaskConfiguration.java]: There is a typo here. This thould be eg.: x28-ontology-xml-import

File: app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/config/X28OntologyImportTaskConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                .incrementer(new RunIdIncrementer())
+                .listener(new CleanupXmlFileJobExecutionListener())
+                .start(stepBuilderFactory
+                        .get("download-from-sftp")
```
Comment (birom) [inline app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/config/X28OntologyImportTaskConfiguration.java]: We also need here .allowStartIfComplete(true).

File: app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/config/X28OntologyImportTaskConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+    @Bean
+    Jaxb2Marshaller X28Marshaller() {
+        Jaxb2Marshaller jaxb2Marshaller = new Jaxb2Marshaller();
+        jaxb2Marshaller.setClassesToBeBound(Row.class);
+        return jaxb2Marshaller;
+    }
+
+    @Bean
+    X28OntologyWriter x28JobAdWriter() {
```
Comment (birom) [inline app-x28-ontology-import-task/src/main/java/ch/admin/seco/service/reference/integration/x28/ontology/importer/config/X28OntologyImportTaskConfiguration.java]: Copy paste error. Use: x28OntologyWriter

**Comment by cjpsycos** [comment]
no more needed

**Pull request closed without merging**